ion/type.go

```diff
@@ -90,6 +90,16 @@ func (t Type) String() string {
 	}
 }
 
+// IsScalar determines if the type is a scalar type
+func (t Type) IsScalar() bool {
```

Comment:
Go tends to prefer top-level functions for these; e.g. `strings.HasPrefix(s, "...")` rather than `s.HasPrefix("...")`. That'd be my recommendation (which you're free to ignore if you prefer it this way :)).

Reply:
Done.